cli.sh

```diff
@@ -1128,99 +1152,104 @@ clear_shared_hook_repos() {
 #   the global or local list, or both.
 #####################################################
 list_shared_hook_repos() {
-    LIST_GLOBAL=1
-    LIST_LOCAL=1
-    LIST_WITH_URL=
+    LIST_SHARED=1
+    LIST_CONFIGS="global,local"
 
     for ARG in "$@"; do
         case "$ARG" in
-        "--global")
-            LIST_LOCAL=
+        "--shared")
+            LIST_CONFIGS=""
             ;;
         "--local")
-            LIST_GLOBAL=
-            ;;
-        "--all")
-            # leave both list options on
+            LIST_CONFIGS="local"
+            LIST_SHARED=""
             ;;
-        "--with-url")
-            LIST_WITH_URL=1
+        "--global")
+            LIST_CONFIGS="global"
+            LIST_SHARED=""
             ;;
+        "--all") ;;
         *)
-            echo "! Unknown list option: $ARG" >&2
+            echo "! Unknown list option \`$ARG\`" >&2
+            echo "  Usage: \`git hooks shared list [--shared|--local|--global|--all]\`" >&2
             exit 1
             ;;
         esac
     done
 
-    if [ -n "$LIST_GLOBAL" ]; then
-        echo "Global shared hook repositories:"
+    IFS="$IFS_COMMA_NEWLINE"
+    for LIST_CONFIG in $LIST_CONFIGS; do
+        unset IFS
+
+        echo "Shared hook repositories in $LIST_CONFIG Git config:"
 
-        if [ -z "$(git config --global --get githooks.shared)" ]; then
+        if [ -z "$(git config "--$LIST_CONFIG" --get githooks.shared)" ]; then
             echo "  - None"
         else
 
             IFS="$IFS_COMMA_NEWLINE"
-            for LIST_ITEM in $(git config --global --get githooks.shared); do
+            for LIST_ITEM in $(git config "--$LIST_CONFIG" --get githooks.shared); do
                 unset IFS
 
                 set_shared_root "$LIST_ITEM"
-                if [ -d "$SHARED_ROOT/.git" ]; then
-                    if [ "$(git -C "$SHARED_ROOT" config --get remote.origin.url)" = "$LIST_ITEM" ]; then
-                        LIST_ITEM_STATE="active"
+
+                LIST_ITEM_STATE="invalid"
+
+                if [ "$SHARED_REPO_IS_CLONED" = "true" ]; then
```

Comment:
Who sets this? Is it handled in `set_shared_root` ?

Reply:
Exactly.